apps/portals/src/portal-components/RedirectDialog.tsx

```diff
@@ -0,0 +1,125 @@
+import * as React from 'react'
+import {
+  Dialog,
+  DialogContent,
+  Button,
+  Link
+} from '@mui/material'
+import { Typography } from 'synapse-react-client'
+
+export type RedirectDialogProps = {
+  onCancelRedirect: ()=>void
+  redirectUrl?: string
+}
+
+export const redirectInstructionsMap = {
+  'https://sites.google.com/sagebase.org/mc2intranet/home?authuser=0' : 
+  <>
+    <Typography variant="body1" sx={{paddingBottom: '20px'}}>
+      <strong>You are currently being redirected to the Multi-Consortia Coordinating (MC2) Center Intranet.</strong>
+    </Typography>
+    <Typography variant="body1" sx={{paddingBottom: '20px'}}>
+      The MC2 Center Intranet is a protected website containing MC2 Center resources. If you are not a participating member of the MC2 Center community, you will not be able to access the intranet.
+    </Typography>
+    <Typography variant="body1" sx={{paddingBottom: '20px'}}>
+      Feel free to <Link target="_blank" rel="noreferrer" href="https://help.cancercomplexity.synapse.org/doc/contact-us">contact the MC2 Center</Link> to learn more, or if you have any questions.
+    </Typography>
+  </>
+}
+
+export const synapseRedirectInstructions =
+  <Typography variant="body1" sx={{paddingBottom: '20px'}}>
+    <strong>You are being redirected to Synapse to view this data.</strong>
+  </Typography>
+
+const isSynapseURL = (url:string) => {
+  if (!url) return false
+  return new URL(url).hostname.toLowerCase() === 'www.synapse.org'
+}
+
+const RedirectDialog = (props: RedirectDialogProps) => {
+  const [countdownSeconds, setCountdownSeconds] = React.useState<number | undefined>()
```

Comment:
Can we inline this so it's never undefined and also remove the effect below?

```suggestion
  const [countdownSeconds, setCountdownSeconds] = React.useState(isSynapseURL(redirectUrl) ? 10 : 30)
```